### PR TITLE
feat: Improving the env vars management in the graphql-server

### PR DESCRIPTION
**Pull request by imrajdas**
Signed-off-by: Raj Babu Das <mail.rajdas@gmail.com>

## Proposed changes

* `os.GetEnv` has been used a lot of times in the codebase. Moving all the envs to EnvConfig makes it easy to identify the vars in one place and easily manageable.
* Move the "cluster/agent" related functions to pkg/cluster.

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [ ] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

File: litmus-portal/graphql-server/server.go [hunk 1]
```diff
@@ -126,7 +87,7 @@ func main() {
 		logrus.Fatal(err)
 	}
 
-	go startGRPCServer(rpcPort) // start GRPC server
+	go startGRPCServer(utils.Config.LitmusAuthGrpcPort) // start GRPC server
```
Comment (gdsoumya) [inline litmus-portal/graphql-server/server.go:90]: I think these are 2 different variables, they might have the same values but we should keep them separate `LitmusAuthGrpcPort` is used to connect to the auth grpc server it seems and `RPC_PORT` one starts the grpc server for graphql

**Review by gdsoumya — CHANGES_REQUESTED** [review]

File: litmus-portal/graphql-server/server.go (hunk 1 above)
Comment (imrajdas) [reply litmus-portal/graphql-server/server.go]: Done

File: litmus-portal/graphql-server/utils/variables.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	WorkflowHelperImageVersion  string `required:"true" split_words:"true"`
+	ServerServiceName           string `split_words:"true"`
+	NodeName                    string `split_words:"true"`
+	Ingress                     string `split_words:"true"`
+	IngressName                 string `split_words:"true"`
+	ChaosCenterUiEndpoint       string `split_words:"true"`
+	TlsCertB64                  string `split_words:"true"`
+	TlsSecretName               string `split_words:"true"`
+	LitmusAuthGrpcEndpoint      string `split_words:"true"`
+	LitmusAuthGrpcPort          string `split_words:"true"`
```
[the remark this answers is not in the document]
Comment (imrajdas) [reply litmus-portal/graphql-server/utils/variables.go]: Done

File: litmus-portal/graphql-server/pkg/k8s/client.go [hunk 3]
```diff
@@ -11,13 +10,13 @@ import (
 )
 
 func GetKubeConfig() (*rest.Config, error) {
-	KubeConfig := os.Getenv("KUBECONFIG")
+	kubeConfig := utils.Config.KubeConfig
```
[the remark this answers is not in the document]
Comment (imrajdas) [reply litmus-portal/graphql-server/pkg/k8s/client.go]: I renamed it to KubeConfgFilePath, as this is not being used anywhere in helm or k8s manifest.

I think this name is an appropriate name for this env.

Other env names are not changed.

[the remark this answers is not in the document]
Comment (imrajdas) [reply litmus-portal/graphql-server/pkg/k8s/client.go]: We are also planning to rename some env to appropriate names in the future.

File: litmus-portal/graphql-server/server.go [hunk 4]
```diff
@@ -105,16 +76,6 @@ func validateVersion() error {
...
-	}
-
-	rpcPort := os.Getenv("RPC_PORT")
-	if rpcPort == "" {
-		rpcPort = utils.DefaultRPCPort
-	}
-
 	client, err := mongodb.MongoConnection()
 	if err != nil {
 		logrus.Fatal(err)
```
Comment (Jonsy13) [inline litmus-portal/graphql-server/server.go:81]: Can you please CodeQL is giving some warning here @imrajdas

**Review by SarthakJain26 — APPROVED** [review]

**Review by Saranya-jena — APPROVED** [review]

**Pull request merged**